Source/WebCore/dom/StaticRange.cpp

```diff
@@ -83,4 +83,19 @@ void StaticRange::visitNodesConcurrently(JSC::AbstractSlotVisitor& visitor) cons
     addWebCoreOpaqueRoot(visitor, end.container.get());
 }
 
+bool StaticRange::computeValidity() const
+{
+    Node& startContainer = this->startContainer();
+    Node& endContainer = this->endContainer();
+
+    if (!connectedInSameTreeScope(&startContainer.rootNode(), &endContainer.rootNode()))
+        return false;
+    if (startOffset() > startContainer.length())
+        return false;
+    if (endOffset() > endContainer.length())
+        return false;
+    if (endOffset() < startOffset() && &startContainer == &endContainer)
```

Comment:
I guess this could be just:
if (&startContainer == &endContainer)
    return endOffset() < startOffset();

Reply:
Logically yes. Will try, If it works I'll push it.